test/functionapp.test.ts

```diff
@@ -0,0 +1,125 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { ResourceManagementClient } from 'azure-arm-resource';
+import { WebSiteManagementClient, WebSiteManagementModels } from 'azure-arm-website';
+import { IHookCallbackContext, ISuiteCallbackContext } from 'mocha';
+import * as vscode from 'vscode';
+import { AzureTreeDataProvider, DialogResponses, ext, extensionPrefix, FunctionAppProvider, getGlobalFuncExtensionSetting, getRandomHexString, ProjectLanguage, projectLanguageSetting, TestAzureAccount, TestUserInput, updateGlobalSetting } from '../extension.bundle';
+import { longRunningTestsEnabled } from './global.test';
+
+// tslint:disable-next-line:max-func-body-length
+suite('Function App actions', async function (this: ISuiteCallbackContext): Promise<void> {
+    this.timeout(1200 * 1000);
+    const resourceGroupsToDelete: string[] = [];
+    let oldAdvancedCreationSetting: boolean | undefined;
+    let oldProjectLanguage: ProjectLanguage | undefined;
+    const resourceName1: string = getRandomHexString().toLowerCase();
+    const testAccount: TestAzureAccount = new TestAzureAccount();
+
+    suiteSetup(async function (this: IHookCallbackContext): Promise<void> {
+        if (!longRunningTestsEnabled) {
```

Comment:
There's a lot of duplicate code in this file and `createAzureResource.test.ts` (at least suiteSetup and suiteTeardown). Please add your tests to the other file instead of copying this code

Reply:
Do you mean that add all our tests from functionapp.test.ts to your createAzureResource.test.ts? Such as stopFunctionApp, startFunctionApp and so on.

Reply:
Yes that is correct. We might rename `createAzureResource.test.ts` to be more generic, but I would prefer to do that after your PR because renaming the file can make it hard to review a PR.